service/controller/resource/app/desired.go

```diff
@@ -240,6 +240,11 @@ func (r *Resource) newAppSpecs(ctx context.Context, cr v1alpha1.ClusterGuestConf
 			}
 		}
 
+		// legacy cluster requires to link nginx with a custom configmap.
```

Comment:
```suggestion
		// Nginx Ingress Controller uses its own configmap that includes the number of workers.
```

Reply:
number of workers is just one of many properties configurable through nginx configmap - what's special about number of workers to be mentioned in the comment?

Reply:
The problem is dynamic values. The values in the *-cluster-values configmap need to be static for the lifetime of the cluster.

But the worker count can change when the cluster is scaled. Any apps using that configmap would also be updated and trigger a new helm release which we don't want.

So the ingress-controller-values configmap should be the same apart from the worker count. This is just a workaround to the real fix you're working on to enable HPA by default.

Reply:
go with this comment